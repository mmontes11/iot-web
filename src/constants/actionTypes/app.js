export const TOGGLE_HAMBURGER_MENU = "TOGGLE_HAMBURGER_MENU";
export const TOGGLE_MAP_DIALOG = "TOGGLE_MAP_DIALOG";
