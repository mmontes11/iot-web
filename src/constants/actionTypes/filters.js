export const FILTER_TYPE_SELECT = "FILTER_TYPE_SELECT";
export const ADD_FILTER_TYPE = "ADD_FILTER_TYPE";
