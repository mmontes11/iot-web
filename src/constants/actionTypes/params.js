export const PARAM_SELECT = "PARAM_SELECT";
export const PARAM_UPDATED = "PARAM_UPDATED";
export const PARAM_REQUEST = "PARAM_REQUEST";
export const PARAM_REQUEST_SUCCESS = "PARAM_REQUEST_SUCCESS";
export const PARAM_REQUEST_ERROR = "PARAM_REQUEST_ERROR";
export const PARAM_ITEMS_UPDATED = "PARAM_ITEMS_UPDATED";
