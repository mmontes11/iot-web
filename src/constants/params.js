export const FIRST_PARAM = "FIRST_PARAM";
export const SECOND_PARAM = "SECOND_PARAM";
