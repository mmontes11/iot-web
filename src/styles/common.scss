html {
  overflow: auto;
  background-color: $light;
}

.is-horizontal-center {
  display: flex;
  justify-content: center;
}

.is-selected {
  background-color: $selected;
}
