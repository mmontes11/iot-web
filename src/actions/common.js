import { RESET } from "constants/actionTypes/common";

export const reset = () => dispatch => {
  dispatch({ type: RESET });
};
