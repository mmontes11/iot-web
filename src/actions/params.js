import {
  PARAM_SELECT,
  PARAM_UPDATED,
  PARAM_REQUEST,
  PARAM_REQUEST_SUCCESS,
  PARAM_REQUEST_ERROR,
  PARAM_ITEMS_UPDATED,
} from "constants/actionTypes/params";
import { EVENT_TYPE, MEASUREMENT_TYPE, OBSERVATION_TYPES } from "constants/observationTypes";
import { FIRST_PARAM, SECOND_PARAM } from "constants/params";
import iotClient from "lib/iotClient";

const requestTypes = (type, onStart, onSuccess, onError) => {
  if (type === EVENT_TYPE) {
    onStart();
    iotClient.eventService
      .getTypes()
      .then(res => onSuccess(res))
      .catch(err => onError(err));
  } else if (type === MEASUREMENT_TYPE) {
    onStart();
    iotClient.measurementService
      .getTypes()
      .then(res => onSuccess(res))
      .catch(err => onError(err));
  }
};

export const selectType = () => dispatch => {
  dispatch({ type: PARAM_ITEMS_UPDATED, param: FIRST_PARAM, items: OBSERVATION_TYPES });
  dispatch({ type: PARAM_SELECT, param: FIRST_PARAM });
};

export const updateType = type => dispatch => {
  dispatch({ type: PARAM_UPDATED, param: FIRST_PARAM, selectedItem: type });
  requestTypes(
    type,
    () => dispatch({ type: PARAM_REQUEST, param: SECOND_PARAM }),
    res => {
      dispatch({ type: PARAM_REQUEST_SUCCESS, param: SECOND_PARAM, statusCode: res.statusCode, error: null });
      dispatch({ type: PARAM_ITEMS_UPDATED, param: SECOND_PARAM, items: res.body.types });
    },
    error => {
      dispatch({ type: PARAM_REQUEST_ERROR, param: SECOND_PARAM, statusCode: error.statusCode, error });
    },
  );
};

export const selectObservation = () => dispatch => {
  dispatch({ type: PARAM_SELECT, param: SECOND_PARAM });
};

export const updateObservation = observation => dispatch => {
  dispatch({ type: PARAM_UPDATED, param: SECOND_PARAM, selectedItem: observation });
};

export const updateParams = (firstParam, secondParam) => dispatch => {
  dispatch({ type: PARAM_UPDATED, param: FIRST_PARAM, selectedItem: firstParam });
  dispatch({ type: PARAM_UPDATED, param: SECOND_PARAM, selectedItem: secondParam });
};
