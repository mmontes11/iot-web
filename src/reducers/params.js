import {
  PARAM_SELECT,
  PARAM_UPDATED,
  PARAM_REQUEST,
  PARAM_REQUEST_SUCCESS,
  PARAM_REQUEST_ERROR,
  PARAM_ITEMS_UPDATED,
} from "constants/actionTypes/params";
import { RESET } from "constants/actionTypes/common";
import { FIRST_PARAM, SECOND_PARAM } from "constants/params";

const paramInitialState = {
  items: [],
  isActive: false,
  isLoading: false,
  isDisabled: true,
  selectedItem: null,
};

export const initialState = {
  [FIRST_PARAM]: paramInitialState,
  [SECOND_PARAM]: paramInitialState,
  reset: {
    isDisabled: true,
  },
};

export default (state = initialState, { type, param: paramName, selectedItem, items }) => {
  if (type === RESET) {
    return initialState;
  }
  if (!paramName) {
    return state;
  }
  const param = state[paramName];
  switch (type) {
    case PARAM_SELECT:
      return { ...state, [paramName]: { ...param, isActive: !param.isActive } };
    case PARAM_UPDATED: {
      let newState = {
        ...state,
        [paramName]: { ...param, isActive: false, selectedItem },
        reset: { ...state.reset, isDisabled: false },
      };
      if (paramName === FIRST_PARAM) {
        newState = {
          ...newState,
          [SECOND_PARAM]: paramInitialState,
        };
      }
      return newState;
    }
    case PARAM_REQUEST:
      return { ...state, [paramName]: { ...param, isLoading: true, isDisabled: true } };
    case PARAM_REQUEST_SUCCESS:
      return { ...state, [paramName]: { ...param, isLoading: false, isDisabled: false } };
    case PARAM_REQUEST_ERROR:
      return { ...state, [paramName]: { ...param, isLoading: false, isDisabled: true } };
    case PARAM_ITEMS_UPDATED:
      return { ...state, [paramName]: { ...param, items } };
    default:
      return state;
  }
};

export const getFirstParam = state => state[FIRST_PARAM];
export const getSecondParam = state => state[SECOND_PARAM];
