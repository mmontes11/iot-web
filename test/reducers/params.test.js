import deepFreeze from "deep-freeze";
import paramsReducer, { initialState as paramsInitialState } from "reducers/params";
import {
  PARAM_SELECT,
  PARAM_UPDATED,
  PARAM_REQUEST,
  PARAM_REQUEST_SUCCESS,
  PARAM_REQUEST_ERROR,
  PARAM_ITEMS_UPDATED,
} from "constants/actionTypes/params";
import { RESET } from "constants/actionTypes/common";
import { FIRST_PARAM, SECOND_PARAM } from "constants/params";

deepFreeze(paramsInitialState);

describe("reducers/params", () => {
  it("has a default state", () => {
    expect(paramsReducer(undefined, { type: "@init" })).toMatchSnapshot();
  });
  it("reduces nothing", () => {
    expect(paramsReducer(paramsInitialState, { type: "WHATEVER" })).toMatchSnapshot();
  });
  it("reduces PARAM_SELECT", () => {
    expect(paramsReducer(paramsInitialState, { type: PARAM_SELECT, param: FIRST_PARAM })).toMatchSnapshot();
  });
  it("updates FIRST_PARAM", () => {
    expect(
      paramsReducer(paramsInitialState, { type: PARAM_UPDATED, param: FIRST_PARAM, selectedItem: "foo" }),
    ).toMatchSnapshot();
  });
  it("updates SECOND_PARAM", () => {
    expect(
      paramsReducer(paramsInitialState, { type: PARAM_UPDATED, param: SECOND_PARAM, selectedItem: "bar" }),
    ).toMatchSnapshot();
  });
  it("reduces PARAM_REQUEST", () => {
    expect(paramsReducer(paramsInitialState, { type: PARAM_REQUEST, param: FIRST_PARAM })).toMatchSnapshot();
  });
  it("reduces PARAM_REQUEST_SUCCESS", () => {
    expect(paramsReducer(paramsInitialState, { type: PARAM_REQUEST_SUCCESS, param: FIRST_PARAM })).toMatchSnapshot();
  });
  it("reduces PARAM_REQUEST_ERROR", () => {
    expect(paramsReducer(paramsInitialState, { type: PARAM_REQUEST_ERROR, param: FIRST_PARAM })).toMatchSnapshot();
  });
  it("reduces PARAM_ITEMS_UPDATED", () => {
    expect(
      paramsReducer(paramsInitialState, {
        type: PARAM_ITEMS_UPDATED,
        param: FIRST_PARAM,
        items: [],
      }),
    ).toMatchSnapshot();
  });
  it("reduces RESET", () => {
    expect(paramsReducer(paramsInitialState, { type: RESET })).toMatchSnapshot();
  });
});
