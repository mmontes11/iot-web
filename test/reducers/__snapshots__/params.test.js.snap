// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers/params has a default state 1`] = `
Object {
  "FIRST_PARAM": Object {
    "isActive": false,
    "isDisabled": true,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
  "SECOND_PARAM": Object {
    "isActive": false,
    "isDisabled": true,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
  "reset": Object {
    "isDisabled": true,
  },
}
`;

exports[`reducers/params reduces PARAM_ITEMS_UPDATED 1`] = `
Object {
  "FIRST_PARAM": Object {
    "isActive": false,
    "isDisabled": true,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
  "SECOND_PARAM": Object {
    "isActive": false,
    "isDisabled": true,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
  "reset": Object {
    "isDisabled": true,
  },
}
`;

exports[`reducers/params reduces PARAM_REQUEST 1`] = `
Object {
  "FIRST_PARAM": Object {
    "isActive": false,
    "isDisabled": true,
    "isLoading": true,
    "items": Array [],
    "selectedItem": null,
  },
  "SECOND_PARAM": Object {
    "isActive": false,
    "isDisabled": true,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
  "reset": Object {
    "isDisabled": true,
  },
}
`;

exports[`reducers/params reduces PARAM_REQUEST_ERROR 1`] = `
Object {
  "FIRST_PARAM": Object {
    "isActive": false,
    "isDisabled": true,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
  "SECOND_PARAM": Object {
    "isActive": false,
    "isDisabled": true,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
  "reset": Object {
    "isDisabled": true,
  },
}
`;

exports[`reducers/params reduces PARAM_REQUEST_SUCCESS 1`] = `
Object {
  "FIRST_PARAM": Object {
    "isActive": false,
    "isDisabled": false,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
  "SECOND_PARAM": Object {
    "isActive": false,
    "isDisabled": true,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
  "reset": Object {
    "isDisabled": true,
  },
}
`;

exports[`reducers/params reduces PARAM_SELECT 1`] = `
Object {
  "FIRST_PARAM": Object {
    "isActive": true,
    "isDisabled": true,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
  "SECOND_PARAM": Object {
    "isActive": false,
    "isDisabled": true,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
  "reset": Object {
    "isDisabled": true,
  },
}
`;

exports[`reducers/params reduces RESET 1`] = `
Object {
  "FIRST_PARAM": Object {
    "isActive": false,
    "isDisabled": true,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
  "SECOND_PARAM": Object {
    "isActive": false,
    "isDisabled": true,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
  "reset": Object {
    "isDisabled": true,
  },
}
`;

exports[`reducers/params reduces nothing 1`] = `
Object {
  "FIRST_PARAM": Object {
    "isActive": false,
    "isDisabled": true,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
  "SECOND_PARAM": Object {
    "isActive": false,
    "isDisabled": true,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
  "reset": Object {
    "isDisabled": true,
  },
}
`;

exports[`reducers/params updates FIRST_PARAM 1`] = `
Object {
  "FIRST_PARAM": Object {
    "isActive": false,
    "isDisabled": true,
    "isLoading": false,
    "items": Array [],
    "selectedItem": "foo",
  },
  "SECOND_PARAM": Object {
    "isActive": false,
    "isDisabled": true,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
  "reset": Object {
    "isDisabled": false,
  },
}
`;

exports[`reducers/params updates SECOND_PARAM 1`] = `
Object {
  "FIRST_PARAM": Object {
    "isActive": false,
    "isDisabled": true,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
  "SECOND_PARAM": Object {
    "isActive": false,
    "isDisabled": true,
    "isLoading": false,
    "items": Array [],
    "selectedItem": "bar",
  },
  "reset": Object {
    "isDisabled": false,
  },
}
`;
